spec/requests/authentication_pages_spec.rb

```diff
@@ -30,16 +29,77 @@
 
     describe 'with valid information' do
       let(:user) { FactoryGirl.create(:user) }
-
-      before { valid_signin(user) }
+      before { sign_in user }
 
       it { should have_title(user.name) }
-      it { be_signin_header }
+      it { should have_link('Profile',     href: user_path(user)) }
+      it { should have_link('Settings',    href: edit_user_path(user)) }
+      it { should have_link('Sign out',    href: signout_path) }
+      it { should_not have_link('Sign in', href: signin_path) }
 
       describe 'folllowd by signout' do
         before { click_link 'Sign out' }
-        it { should have_link('Sign in') }
+
+        it { should_not have_title(user.name) }
+        it { should_not have_link('Profile',   href: user_path(user)) }
+        it { should_not have_link('Settings',  href: edit_user_path(user)) }
+        it { should_not have_link('Sign out',  href: signout_path) }
+        it { should have_link('Sign in',       href: signin_path) }
+      end
+    end
+  end
+
+  describe "authorization" do
+    describe "for non-signed-in users" do
+      let(:user) { FactoryGirl.create(:user) }
+
+      describe "when attempting to visit a protected page" do
+        before do
+          visit edit_user_path(user)
+          fill_in "Email",    with: user.email
+          fill_in "Password", with: user.password
+          click_button "Sign in"
+        end
+        
+        describe "after signing in" do
```

Comment:
Prefer single-quoted strings when you don't need string interpolation or special symbols.